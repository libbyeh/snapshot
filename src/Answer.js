import React, { Component } from 'react';
import './styles/main.scss';
import Flashcards from './Flashcards';
import App from './App'

class Answer extends Component {
  constructor(props) {
    super();
    }

  saveFlashcard = () => {
    const savedCard = JSON.parse(localStorage.getItem('savedCard'))
    if (!savedCard) {
      localStorage.setItem('savedCard', JSON.stringify([this.props.flashcardObject]))
    } else {
      savedCard.push(this.props.flashcardObject)
      localStorage.setItem('savedCard', JSON.stringify(savedCard))
    }
  }

  render() {
    if (!this.props.result) {
      return ( <div className='none'></div> );
    } else {
    return (
      <div className='answer-popup'>
        <div className='answer-body'>
          <div className='answer-polaroid-image-pop'>
            <div className='answer-box'>
              <h2 className='answer-flashcard-option-title'>Guess a Prototype Method</h2>
              <h3 className='correct-answer' >{this.props.result}</h3>
              <h3 className='answer-flashcard' >Correct Answer:  {this.props.flashcardObject.answer}</h3>
              <button className='back-flashcard-button' onClick={this.props.nextQuestion}>Next Question</button>
              <a className='flashcard-info-link' href={this.props.flashcardObject.link}>Learn More</a>
              <button className='save-button' onClick={this.saveFlashcard}>I Need More Practice - Save Concept</button>
            </div>
          </div>
        </div>
      </div>
    )
    }
  }
}


export default Answer;