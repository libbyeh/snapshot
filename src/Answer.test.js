import React from 'react'
import Answer from './Answer'
import { shallow } from 'enzyme'


describe ('Answer', () => {
  let wrapper = shallow(
    <Answer  />
    )

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  })

})