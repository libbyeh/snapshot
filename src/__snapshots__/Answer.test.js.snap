// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should match the snapshot with all data passed in 1`] = `
<div
  className="none"
/>
`;
